src/main/java/io/vertx/core/net/impl/pool/SimpleConnectionPool.java

```diff
@@ -194,11 +201,14 @@ public long concurrency() {
   private final Waiters<C> waiters;
   private int requests;
 
-  SimpleConnectionPool(PoolConnector<C> connector, int[] maxSizes) {
-    this(connector, maxSizes, -1);
+  private final java.util.concurrent.Executor resumeExecutor;
```

Comment:
in practice this will not work because executor is the current event-loop.

since this class is already married to the `EventLoopContext`, we should assume it and replace the `resumeExecutor` with the current `EventLoopContext#executor()`

Reply:
Perfect, will change it :+1:

Reply:
Looking at it twice I don't see how:
> since this class is already married to the EventLoopContext

EventLoopContext is used on the Slot by using
```java
EventLoopContext connectionContext = pool.contextProvider.apply(waiter.context);
```
Given that I dont't have a `ContextInternal` (the type of `waiter,context`) how can I obtain it on pool creation and/or
submit/resume?